### PR TITLE
[ZEPPELIN-1180] Update publish_release.sh to publish scala 210 and scala 211 artifacts

### PR DESCRIPTION
**Pull request by minahlee**
### What is this PR for?
- Update publish_release.sh to publish scala 2.10 and scala 2.11 artifacts to maven repository.
- Set build spark version to 2.0

This PR supposed to work once #1186 and #1195 is merged

### What type of PR is it?
Improvement

### What is the Jira issue?
[ZEPPELIN-1180](https://issues.apache.org/jira/browse/ZEPPELIN-1180)

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: dev/publish_release.sh [hunk 1]
```diff
@@ -92,13 +93,27 @@ function publish_to_maven() {
...
+    exit 1
+  fi
+
+  # build with scala-2.11
+  "${BASEDIR}/change_scala_version.sh" 2.11
+
+  echo "mvn clean install -Ppublish-distr \
+    -Dmaven.repo.local=${tmp_repo} -Dscala-2.11 \
+    ${PUBLISH_PROFILES} ${PROJECT_OPTIONS}"
+  mvn clean install -Ppublish-distr -Dmaven.repo.local="${tmp_repo}" -Dscala-2.11\
```
Comment (lresende) [inline dev/publish_release.sh]: to be consistent with other places such as travis ci config, I would use p_scalal-2.11

File: dev/publish_release.sh [hunk 2]
```diff
@@ -44,8 +44,9 @@ NC='\033[0m' # No Color
 RELEASE_VERSION="$1"
 GIT_TAG="$2"
 
-PUBLISH_PROFILES="-Pspark-1.6 -Phadoop-2.4 -Pyarn -Ppyspark -Psparkr -Pr"
+PUBLISH_PROFILES="-Pspark-2.0 -Phadoop-2.4 -Pyarn -Ppyspark -Psparkr -Pr"
 PROJECT_OPTIONS="-pl !zeppelin-distribution"
+PROJECT_OPTIONS_SCALA211="-pl zeppelin-interpreter,cassandra,flink,ignite,spark,spark-dependencies,r,zeppelin-display"
```
Comment (lresende) [inline dev/publish_release.sh]: I would recommend though that we do a clean and a full recompile with scala 2.11 to be on the safe mode but
if we are going to build projects individually, we need to include scalding, zeppelin_server and even maybe zeppelin-distribution

grep -r --include=pom.xml "_2.10</" * 
cassandra/pom.xml:    <artifactId>zeppelin-cassandra_2.10</artifactId>
flink/pom.xml:  <artifactId>zeppelin-flink_2.10</artifactId>
ignite/pom.xml:  <artifactId>zeppelin-ignite_2.10</artifactId>
r/pom.xml:  <artifactId>zeppelin-zrinterpreter_2.10</artifactId>
scalding/pom.xml:  <artifactId>zeppelin-scalding_2.10</artifactId>
spark/pom.xml:  <artifactId>zeppelin-spark_2.10</artifactId>
spark-dependencies/pom.xml:  <artifactId>zeppelin-spark-dependencies_2.10</artifactId>
zeppelin-display/pom.xml:  <artifactId>zeppelin-display_2.10</artifactId>
zeppelin-distribution/pom.xml:      <artifactId>zeppelin-server_2.10</artifactId>
zeppelin-server/pom.xml:  <artifactId>zeppelin-server_2.10</artifactId>

**Comment by minahlee** [comment]
@lresende thank you for review. I addressed your comments in 1a82dc5

**Comment by lresende** [comment]
LGTM

**Comment by minahlee** [comment]
Merge if there is no more discussion

**Pull request closed without merging**